### ISSUE TITLE
Failed to start database: SQL module was unable to write a schema file to the current temp directory

### ISSUE THREAD

**Issue by malx122** [Starcounter/Home#77]
I have started to have trouble starting a database on my developer machine. I get:
```
[13:18:34, Critical, Starcounter (david)]
System.Exception: ScErrSqlExportSchemaFailed (SCERR7001): SQL module was unable to write a schema file to the current temp directory. Failed to load SQL schema info from file: C:/Users/david/Documents/Starcounter/Personal/Temp/david/david-20170405T095246829/sqlschemas/schema170405131828338704371665.pl. Error message: Unable to read data from the transport connection: A blocking operation was interrupted by a call to WSACancelBlockingCall.
Version: 2.3.0.6122.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR7001.
   at Starcounter.Query.Sql.PrologManager.ExportSchemaInfo(Scheduler scheduler, String fullAppId, TypeDef[] typeDefArray, Boolean useFullNamespaceOnly) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Starcounter\Query\SQL\PrologManager.cs:line 113
   at Starcounter.Query.QueryModule.UpdateSchemaInfo(String fullAppId, TypeDef[] typeDefArray, Boolean useFullNamespaceOnly) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Starcounter\Query\_QueryModule.cs:line 87
   at Starcounter.Hosting.Package.UpdateDatabaseSchemaAndRegisterTypes(String fullAppId, TypeDef[] unregisteredTypeDefs, TypeDef[] allTypeDefs) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Hosting\Starcounter.Hosting\Package.cs:line 413
   at Starcounter.Hosting.Package.ProcessWithinCurrentApplication(Application application, ApplicationDirectory applicationDir) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Hosting\Starcounter.Hosting\Package.cs:line 265
   at Starcounter.Hosting.Package.Process() in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Hosting\Starcounter.Hosting\Package.cs:line 138
   at Starcounter.Hosting.Package.Process(IntPtr hPackage) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Hosting\Starcounter.Hosting\Package.cs:line 55
   at StarcounterInternal.Hosting.Processor.RunMessageLoop(Void* hsched) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Hosting\Starcounter.Hosting\Processor.cs:line 85
HResult=-2146233088
HelpLink=https://github.com/Starcounter/Starcounter/wiki/SCERR7001
---> System.IO.IOException: Unable to read data from the transport connection: A blocking operation was interrupted by a call to WSACancelBlockingCall.
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   at System.IO.BufferedStream.ReadByte()
   at se.sics.prologbeans.FastParser.parseProlog(Stream stream) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\bin\Release\FastParser.cs:line 83
   at se.sics.prologbeans.PrologSession.send(String query, Bindings bindings, String sessionID) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\bin\Release\PrologSession.cs:line 550
   at se.sics.prologbeans.PrologSession.executeQuery(String query, Bindings bindings) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\bin\Release\PrologSession.cs:line 455
   at Starcounter.Query.Sql.PrologManager.ExecuteQuery(PrologSession session, String query, Bindings bindings) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Starcounter\Query\SQL\PrologManager.cs:line 348
   at Starcounter.Query.Sql.PrologManager.CallSqlProcessToLoadSchemaInfo(Scheduler scheduler, String schemaFilePath) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Starcounter\Query\SQL\PrologManager.cs:line 933
   at Starcounter.Query.Sql.PrologManager.ExportSchemaInfo(Scheduler scheduler, String fullAppId, TypeDef[] typeDefArray, Boolean useFullNamespaceOnly) in C:\TeamCity\BuildAgent\work\sc-4138\Level1\src\Starcounter\Query\SQL\PrologManager.cs:line 107
HResult=-2146232800
---> System.Net.Sockets.SocketException: A blocking operation was interrupted by a call to WSACancelBlockingCall
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
ErrorCode=10004
NativeErrorCode=10004
HResult=-2147467259
```

**Comment by malx122** [comment]
Started by happening maybe each third-fourth restart 3 days ago. After that more and more often. Now I might need to restart the db 4-5 times before I succeed.

**Comment by miyconst** [comment]
cc @k-rus, @bigwad

**Comment by miyconst** [comment]
@malx122 thank you for reporting.

**Comment by k-rus** [comment]
cc @alemoi

**Comment by alemoi** [comment]
No changes in this area for a really long time. Something in 3rd party `se.sics.prologbeans`.

**Comment by k-rus** [comment]
> Something in 3rd party se.sics.prologbeans.

@alemoi Do you mean that someone did changes there?

**Comment by alemoi** [comment]
No, I mean maybe there is a bug in this 3rd party component.

**Comment by alemoi** [comment]
@malx122 how often do you get this one?

**Comment by malx122** [comment]
Like I said, now every time I try to restart the apps, up to 4-5 times before I successfully start the db.

**Comment by alemoi** [comment]
I'm sorry, I missed your post. 
Not sure if this might help but try to clean the directory: `C:/Users/david/Documents/Starcounter/Personal/Temp`

Also it even could be that some software blocks the sockets for Prolog? Maybe an antivirus or something?

**Comment by miyconst** [comment]
Archiving the issue.

**Comment by atifwaqar** [comment]
We got hit by this issue out of no where on one of our live server. Any idea why this issue occurs? and how to prevent this issue in future? 

@alemoi Thanks for the solution, your provided solution fixed the issue for me :)

**Comment by ubbeK** [comment]
I got hit by this issue today, restarting Starcounter solved it.

Starcounter version: `2.3.1.8415`
OS: Windows Server 2019 as VM
Running starcounter as a service.

**Comment by miyconst** [comment]
The issue is fixed in 2.4, but we still shall fix it in 2.3.x.

**Comment by ubbeK** [comment]
I haven't been hit by this since my last comment. Has this been solved in `2.3.1.8584`?

**Comment by miyconst** [comment]
As far as I know, it was not fixed in 2.3.x.

**Comment by miyconst** [comment]
Closing as won't fix in Starcounter 2 and the issue is not applicable for the upcoming Starcounter 3.
